src/test/kotlin/de/bund/digitalservice/useid/statics/WidgetControllerIntegrationTest.kt

```diff
@@ -10,14 +10,44 @@ import org.springframework.test.web.reactive.server.WebTestClient
 class WidgetControllerIntegrationTest(@Autowired val webTestClient: WebTestClient) : PostgresTestcontainerIntegrationTest() {
 
     @Test
-    fun `should disable X-Frame-Options`() {
+    fun `widget endpoint should disable X-Frame-Options`() {
         webTestClient
             .get()
-            .uri("/widget")
+            .uri("/widget?hostname=foo.bar")
             .exchange()
             .expectStatus()
             .isOk
             .expectHeader()
             .doesNotExist("X-Frame-Options")
     }
+
+    @Test
+    fun `widget endpoint should deliver Content-Security-Policy with allowed host when the request URL is valid`() {
+        webTestClient
+            .get()
+            .uri("/widget?hostname=foo.bar")
+            .exchange()
+            .expectStatus()
+            .isOk
+            .expectHeader()
+            .valueEquals(
+                "Content-Security-Policy",
+                "some default value;frame-ancestors 'self' foo.bar;"
+            )
+    }
+
+    @Test
+    fun `widget endpoint should deliver default Content-Security-Policy when the request URL is invalid`() {
```

Comment:
Maybe we should add a similar test for the case that the parameter is not set at all?